packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts

```diff
@@ -714,7 +714,6 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
     return (
       Object.keys(grouped)
         .map(typeName => {
-          const hasUnions = grouped[typeName].filter(s => typeof s !== 'string' && s.union).length > 0;
           const relevant = grouped[typeName].filter(Boolean);
 
           if (relevant.length === 0) {
```

Comment:
`relevant.length === 0` will be `true` for `[null]` so it exists before declaring `hasUnions`.